### ISSUE TITLE
Add a Web based data source

### ISSUE THREAD

**Issue by crutkas** [microsoft/WindowsTemplateStudio#133]
Syncing with Unni, VS has a todo item with Azure.  This doesn't require the Azure SDK installed evidently.

**Comment by mrlacey** [comment]
Implementation plan here is to include a WebAPI project in the solution and connect to that. Very similar to what Xamarin templates do.

**Comment by mrlacey** [comment]
Requires multi-project support first

**Comment by mrlacey** [comment]
Moved to 2.3 as this should really be done after we move to multi-project solutions and that won't happen until 2.3 at the earliest.

**Comment by mrlacey** [comment]
Initial thoughts
- Base this on the output of a Xamarin.Forms app which includes an 'ASP.NET Core Web API project'.
- Have the generated project be called `{App}.AppService` (Drop the "Mobile".)
- Have the current SampleDataService implement a new `IDataService` interface.
- Have this new interface define methods that are also implemented by a new `WebDataService` that can also return, by default, the same data (or very similar) as the SampleDataService.
- Review how this new interface will work with Prism apps that already include a `ISampleDataService`.
- Need to investigate RightClick options. Will it be possible to update the interface accordingly if an app is created with a page using sample data and the WebAPI project is then added by RightClick. Possible same issue for apps created with earlier versions of templates.
- This will be different to XF as that includes CRUD operations on items in the list.
- The XF app also includes item detail and add item pages. This won't as there are already WTS pages that show data. Having an "Add" page would also require validation and this is blocked by the same reasons as #291 & #2293 
- WebAPI project to also reference the core project so it can use the exact same models.

**Comment by mrlacey** [comment]
To include the WebAPI project needs the appropriate VS workload installed. Will need to add a new way to define this requirement in the template config and only display the option if the workload is installed.

Customizing the Xamarin sample code to include a separate `HttpDataSource` that is a separate template the WebAPI one depends on and can also be used for other endpoints. This will also be available regardless of other installed VS workloads.

**Comment by mrlacey** [comment]
.NET Core doesn't support VB until v3
I'm proposing to have the VB version create a Web API project that targets .NET Framework while the C# version targets .NET Core.

This will require some exceptions in the language syncing and comparison checks.
The alternative is to not have a VB version until it's supported by .NET Core.

**Comment by crutkas** [comment]
i would say lets roll with it in c# then create a new issue to backport once core 3 comes on line for VB

**Comment by mrlacey** [comment]
When the "ASP.NET and web development" workload isn't installed

![image](https://user-images.githubusercontent.com/189547/58550833-930c9200-8206-11e9-91d4-ad18e2bca7c4.png)

Features looks like this

![image](https://user-images.githubusercontent.com/189547/58550738-5b054f00-8206-11e9-838d-aeaa7493f375.png)

When the workload is installed

![image](https://user-images.githubusercontent.com/189547/58550915-be8f7c80-8206-11e9-894e-19202240f0b2.png)

Features include the item that depends on it

![image](https://user-images.githubusercontent.com/189547/58553545-14672300-820d-11e9-9fbc-1287000943db.png)

**Comment by mrlacey** [comment]
Verified in 
Templates version: 0.20.19159.1
Wizard version: 0.20.19159.1
